Support for a remote db config

ttung opened the pull request:
### Description
For the most part, accessing the DB is a function of AWS_PROFILE or AWS_ACCESS_KEY_ID/AWS_SECRET_KEY_ID.  We want to have an object that encompasses all the remote db config so we can access the DB from the CLI.

If we didn't do this, then we would require the user to specify --staging or --prod (to initialize `StagingConfig` or `ProductionConfig`), when it actually doesn't matter.

### Test plan
flask run

phoenixAja commented on src/py/aspen/config/config.py:
```diff
@@ -58,26 +58,29 @@ class Config:
+        if descriptive_name is not None:
+            Config._subclasses[descriptive_name] = cls
+
+            # get a list of @flaskproperties for Config and the subclass of Config, and
+            # compare to see if there are any attributes that were marked as
+            # @flaskproperty in Config but not in the subclass.
```
don't ABC's do something similar to this?

ttung replied:
If you tag something as an abstract class (abc.abstractclass), or there are abstract methods (abc.abstractmethod), it cannot be instantiated.  This is similar in concept but we can't quite use those constructs to achieve the same.

phoenixAja commented on src/py/aspen/config/config.py:
```diff
@@ -187,3 +189,11 @@ def DATABASE_URI(self) -> str:
         raise NotImplementedError()
+
+
+class RemoteConfig(Config):
```
i feel like a more intuitive name might be `RemoteDBConfig`? Unless we're planning on using this config for things other that a DB down the line

ttung replied:
It's subclassed by all configs that use a remote db, so I guess you're right.

ttung replied:
Updated.

phoenixAja commented on src/py/aspen/config/production.py:
```diff
@@ -1,12 +1,10 @@
-class ProductionConfig(config.Config, descriptive_name="prod"):
+class ProductionConfig(config.RemoteConfig, descriptive_name="prod"):
     @config.flaskproperty
-    def SECRET_KEY(self) -> str:
-        return uuid.uuid4().hex
```
do we need to add `SECRET_KEY` to the `RemoteConfig` if we're removing it here?

phoenixAja replied:
ok i see you added this back in later PRs, nevermind

ttung replied:
A terrible implementation is provided by the base class now.

phoenixAja commented on src/py/aspen/config/staging.py:
```diff
@@ -1,29 +1,20 @@
     def DEBUG(self) -> bool:
         return True
 
-    @config.flaskproperty
-    def SECRET_KEY(self) -> str:
-        return uuid.uuid4().hex
-
```
samesies

Commit:
Support for a remote db config

### Description
For the most part, accessing the DB is a function of AWS_PROFILE or AWS_ACCESS_KEY_ID/AWS_SECRET_KEY_ID.  We want to have an object that encompasses all the remote db config so we can access the DB from the CLI.

If we didn't do this, then we would require the user to specify --staging or --prod (to initialize `StagingConfig` or `ProductionConfig`), when it actually doesn't matter.

### Test plan
flask run
